### PR TITLE
libsodium: add v1.0.20

### PR DESCRIPTION
**Pull request by adamjstewart**

### CODE REVIEW

**Review by wdconinc — DISMISSED** [review]
Checksum verified.

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Review by bernhardkaindl — DISMISSED** [review]
built

File: var/spack/repos/builtin/packages/libsodium/package.py [hunk 1]
```diff
@@ -36,7 +37,10 @@ class Libsodium(AutotoolsPackage):
     version("1.0.0", sha256="ced1fe3d2066953fea94f307a92f8ae41bf0643739a44309cbe43aa881dbc9a5")
     version("0.7.1", sha256="ef46bbb5bac263ef6d3fc00ccc11d4690aea83643412919fe15369b9870280a7")
 
-    depends_on("c", type="build")  # generated
+    depends_on("c", type="build")
+
+    # https://github.com/jedisct1/libsodium/issues/1372
+    conflicts("target=aarch64:", when="@1.0.19")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libsodium/package.py:43]: 1.0.18 and 1.0.20 both build fine on macOS arm64, but not 1.0.19. Not sure if the target condition is right or if it also includes darwin and/or apple clang versions. Exact error for me looks like:
```
In file included from crypto_aead/aes256gcm/armcrypto/aead_aes256gcm_armcrypto.c:35:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/arm_neon.h:62828:32: error: always_inline function 'vget_high_p64' requires target feature 'crypto', but would be inlined into function 'vmull_high_p64' that is compiled without support for 'crypto'
 62828 |   __ret = vmull_p64((poly64_t)(vget_high_p64(__p0)), (poly64_t)(vget_high_p64(__p1)));
       |                                ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/arm_neon.h:62828:65: error: always_inline function 'vget_high_p64' requires target feature 'crypto', but would be inlined into function 'vmull_high_p64' that is compiled without support for 'crypto'
 62828 |   __ret = vmull_p64((poly64_t)(vget_high_p64(__p0)), (poly64_t)(vget_high_p64(__p1)));
       |                                                                 ^
warning: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: archive library: .libs/libavx2.a the table of contents is empty (no object file members in the library define global symbols)
2 errors generated.
```

Comment (wdconinc) [reply var/spack/repos/builtin/packages/libsodium/package.py:43]: Do you just want to apply https://github.com/jedisct1/libsodium/commit/bb6e6a4b527194e3ca8e8bc7e96b166b361853e3.patch?full_index=1 and not worry about whether this affects other targets?

If not patching, are you waiting for someone to opine on whether the issue and/or target condition includes darwin and/or apple clang, or are you happy to merge as is?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/libsodium/package.py:43]: I'm happy to merge as is. Conflict is fewer lines of code than patch, and there are newer versions with the patch already anyway.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**